tests/index.test.js

```diff
@@ -0,0 +1,88 @@
+/* global __analytics */
+document.body.innerHTML = '<script></script>';
+let analytics;
+let analyticsModules;
+const testStr = 'test';
+const testObj = { test: 'test' };
+const moduleNames = ['amplitude', 'debug', 'googleTagManager', 'infermedicaAnalytics'];
+Object.keys(__analytics).forEach((singleModule) => {
+  __analytics[singleModule].isEnabled = true;
+});
+
+beforeEach(async () => {
+  jest.resetModules();
+  analyticsModules = await import('../src/modules');
+  analytics = await import('../src/index');
+  analyticsModules.default.forEach((singleModule) => {
+    Object.keys(singleModule).forEach((key) => {
+      if (key === 'name') return;
+      jest.spyOn(singleModule, key);
+    });
+  });
+});
+
+describe('analytics function', () => {
+  describe.each(['trackEvent', 'trackView'])('trackView and trackEvent function', (fn) => {
+    test(`${fn} - run all modules`, () => {
+      analytics.Analytics[fn](testStr, testObj, moduleNames);
+      analyticsModules.default.forEach((singleModule) => {
+        if (singleModule[fn]) {
+          expect(singleModule[fn]).toHaveBeenCalledWith(testStr, testObj);
+        }
+      });
+    });
+    test(`${fn} - run only passed modules`, () => {
```

Comment:
👍  but can see why mockups look better than existing modules. Now, this test is connected to modules.js... if you change something with names then it will be broken, but... why, it should be passes ;)